composer/core/data_spec.py

```diff
@@ -0,0 +1,135 @@
+# Copyright 2021 MosaicML. All Rights Reserved.
+from __future__ import annotations
+
+import collections.abc
+import textwrap
+from typing import TYPE_CHECKING, List, Optional, Sequence
+
+import torch
+
+from composer.utils.iter_helpers import ensure_tuple
+
+if TYPE_CHECKING:
+    from composer.core.types import Batch, DataLoader
+
+
+class DataSpec:
```

Comment:
I think users would prefer to pass the functions directly in `__init__`, or as a dictionary with some default functions, rather than have to learn a new class and its logic.

Reply:
Per my suggestion around taking `dict` as input, is it then possible to remove the `DataSpec` class entirely? I believe this was make the code much more readable and simplified.

Reply:
Same as above comment.